cruise-control-client/cruisecontrolclient/client/cccli.py

```diff
@@ -0,0 +1,353 @@
+#!/usr/bin/env python3
+
+# To be able to easily parse command-line arguments
+import argparse
+
+# To be able to make more precise type hints
+from typing import Dict, Tuple, Type  # noqa
+
+# Convenience functions for displaying the retrieved Response
+from cruisecontrolclient.client.Display import display_response
+
+# To be able to instantiate Endpoint objects
+import cruisecontrolclient.client.Endpoint as Endpoint
+
+# To be able to make requests and get a response given a URL to cruise-control
+from cruisecontrolclient.client.Responder import JSONDisplayingResponderGet, JSONDisplayingResponderPost
+
+# To be able to compose a URL to hand to requests
+from cruisecontrolclient.client.Query import generate_url_from_cc_socket_address
+
+
+def add_add_parameter_argument(p: argparse.ArgumentParser):
+    """
+    This should be used with all cruise-control endpoint parsers, to provide
+    forward compatibility and greater operational flexibility.
+    :param p:
+    :return:
+    """
+    p.add_argument('--add-parameter', '--add-parameters', metavar='PARAM=VALUE',
+                   help="Manually specify one or more parameter and its value in the cruise-control endpoint, like 'param=value'",
```

Comment:
Nit: `one or more parameter and its` -> `one or more parameters and their`

Reply:
Agreed, and updated similar number language in `add_remove_parameter_argument`